src/main/java/org/pac4j/vertx/core/engine/VertxSecurityLogic.java

```diff
@@ -13,16 +13,21 @@
    See the License for the specific language governing permissions and
    limitations under the License.
  */
-package org.pac4j.vertx;
+package org.pac4j.vertx.core.engine;
+
+import org.pac4j.core.engine.DefaultSecurityLogic;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.vertx.VertxProfileManager;
+import org.pac4j.vertx.VertxWebContext;
 
 /**
  * @author Jeremy Prime
  * @since 2.0.0
  */
-public interface TestConstants {
-    // User 1, with no permissions
-    public static final String USER_1 = "testUser1";
-    // User 2, with the permission permission1
-    public static final String USER_2 = "testUser2";
-    public static final String USER_2_PERMISSION = "permission1";
+public class VertxSecurityLogic extends DefaultSecurityLogic<Void, VertxWebContext> {
```

Comment:
Or a factory method for a profile manager in the web context class. I'm not averse to using reflection, though if we also use method handles correctly we can make this more efficient (by avoiding an access lookup every time we construct).